sklearn/tests/test_pipeline.py

```diff
@@ -345,10 +364,80 @@ def test_pipeline_fit_transform():
     assert_array_almost_equal(X_trans, X_trans2)
 
 
+def test_set_pipeline_steps():
+    transft1 = TransfT()
+    transft2 = TransfT()
+    pipeline = Pipeline([('mock', transft1)])
+    assert_true(pipeline.named_steps['mock'] is transft1)
+
+    # Directly setting attr
+    pipeline.steps = [('mock2', transft2)]
+    assert_true('mock' not in pipeline.named_steps)
+    assert_true(pipeline.named_steps['mock2'] is transft2)
+    assert_equal([('mock2', transft2)], pipeline.steps)
+
+    # Using set_params
+    pipeline.set_params(steps=[('mock', transft1)])
+    assert_equal([('mock', transft1)], pipeline.steps)
+
+    # Using set_params to replace single step
+    pipeline.set_params(mock=transft2)
+    assert_equal([('mock', transft2)], pipeline.steps)
+
+    # With invalid data
+    assert_raises(TypeError, pipeline.set_params, steps=[('junk', ())])
+
+
+def test_set_pipeline_step_none():
+    """Test setting Pipeline steps to None"""
```

Comment:
usually we use comments, not docstrings, so we can see the function name in the test output.